modules/logging/src/main/scala/com/evolutiongaming/skafka/producer/LoggingProducer.scala

```diff
@@ -1,51 +1,48 @@
 package com.evolutiongaming.skafka.producer
 
-import com.evolutiongaming.concurrent.CurrentThreadExecutionContext
-import com.evolutiongaming.safeakka.actor.ActorLog
+import cats.implicits._
+import cats.{Monad, MonadError}
+import com.evolutiongaming.catshelper.Log
 import com.evolutiongaming.skafka.{OffsetAndMetadata, ToBytes, Topic, TopicPartition}
 
-import scala.concurrent.Future
 import scala.concurrent.duration.FiniteDuration
-import scala.util.{Failure, Success}
 
 object LoggingProducer {
+  type CanFail[F[_]] = MonadError[F, Throwable]
 
-  def apply(producer: Producer[Future], log: ActorLog): Producer[Future] = {
-    implicit val ec = CurrentThreadExecutionContext
+  def apply[F[_] : CanFail : Monad](producer: Producer[F], log: Log[F]): Producer[F] = {
```

Comment:
- Good question about `Monad`, just `FlatMap` suffices actually, changed like that.

- Consequently, used `ApplicativeError` instead of `MonadError` which costed us some syntax sugar but eliminated the need of helper object.